core/src/main/java/org/elasticsearch/index/query/HasChildQueryBuilder.java

```diff
@@ -151,12 +153,15 @@ public InnerHitBuilder innerHit() {
     }
 
     public HasChildQueryBuilder innerHit(InnerHitBuilder innerHit) {
-        innerHit.setParentChildType(type);
-        innerHit.setQuery(query);
+        innerHit.setParentChildType(type, query);
```

Comment:
and we could also make the `setNestedPath` and `setParentChildType` methods pkg-private?